From a44110f22b87d22d26342093b7694c0f9fcba8a3 Mon Sep 17 00:00:00 2001
From: J. Bruce Fields <bfields@redhat.com>
Date: Mon, 23 Jun 2014 21:49:29 -0400
Subject: [fs] nfsd:Add missing recording of back channel attrs in nfsd4_session

Message-id: <1403560314-32525-28-git-send-email-bfields@redhat.com>
Patchwork-id: 84467
O-Subject: [Patch RHEL7 027/172] Add missing recording of back channel attrs in nfsd4_session
Bugzilla: 1111719
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Kinglong Mee <kinglongmee@gmail.com>

commit 5b6feee9608dce7afd2646f457c93e612526d1d8 forgot
recording the back channel attrs in nfsd4_session.

nfsd just check the back channel attars by check_backchannel_attrs,
but do not  record it in nfsd4_session in the latest kernel.

Signed-off-by: Kinglong Mee <kinglongmee@gmail.com>
Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Upstream: 2ce02b6b6cf3532df143b85a72bacd611a55616a
Bugzilla: 1111719
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/nfsd/nfs4state.c b/fs/nfsd/nfs4state.c
index 299772a..5a9a64e 100644
--- a/fs/nfsd/nfs4state.c
+++ b/fs/nfsd/nfs4state.c
@@ -999,6 +999,8 @@ static void init_session(struct svc_rqst *rqstp, struct nfsd4_session *new, stru
  spin_unlock(&nn->client_lock);
  memcpy(&new->se_fchannel, &cses->fore_channel,
    sizeof(struct nfsd4_channel_attrs));
+ memcpy(&new->se_bchannel, &cses->back_channel,
+   sizeof(struct nfsd4_channel_attrs));
  if (cses->flags & SESSION4_BACK_CHAN) {
   struct sockaddr *sa = svc_addr(rqstp);
   /*
-- 
1.7.1